src/extensions/emulator/specHelper.ts

```diff
@@ -36,7 +36,7 @@ function wrappedSafeLoad(source: string): any {
  * directory that contains one. Throws an error if none is found.
  * @param directory the directory to start from searching from.
  */
-export async function findExtensionYaml(directory: string): Promise<string> {
+export function findExtensionYaml(directory: string): Promise<string> {
```

Comment:
This function could be synchronous and return just `string`

Reply:
Sure, though I'm not taking on any more of these requests (it always requires 2 or more changes to fix this kind of thing which will bloat this PR - design issues can be fixed outside of this change)